fili-core/src/test/groovy/com/yahoo/bard/webservice/data/PartialDataHandlerSpec.groovy

```diff
@@ -67,30 +62,8 @@ class PartialDataHandlerSpec extends Specification {
         dimensionDictionary = new DimensionDictionary([dim1, dim2, dim3] as Set)
 
         // Setup mock Request
-        apiRequest = Mock(DataApiRequest.class)
-        apiRequest.getDimensions() >> { [dim1, dim2] as Collection }
-        apiRequest.getFilterDimensions() >> { [] as Set }
-        apiRequest.getFilters() >> { [:] }
-
-        // setup mock inner query
-        GroupByQuery innerQuery = Mock(GroupByQuery.class)
-        innerQuery.getDataSource() >> {
-            new TableDataSource(
-                    new ConcretePhysicalTable(
-                            "basefact_network", DAY.buildZonedTimeGrain(UTC),
-                            [] as Set,
-                            ["userDeviceType": "user_device_type"]
-                    )
-            )
-        }
-        innerQuery.getDependentFieldNames() >> { ["page_views"] as Set }
-
-        groupByQuery.getGranularity() >> WEEK
-        groupByQuery.getIntervals() >> { [new Interval("2014-06-30/2014-07-28")] as List }
-        groupByQuery.getDataSource() >> { new QueryDataSource(innerQuery) }
-        groupByQuery.getDependentFieldNames() >> { innerQuery.getDependentFieldNames() }
-        groupByQuery.getInnermostQuery() >> { innerQuery }
-        groupByQuery.getMetricDimensions() >> []
+        dataSourceConstraint = Stub(QueryPlanningConstraint)
+        dataSourceConstraint.getAllColumnNames() >> columnNames
 
```

Comment:
If you don't care about cardinality on calls, a `Stub` is better, since it allows for _just_ the behavior stubbing parts, w/o asserting anything about how much it's called. Not having assertions that the test doesn't really care about allows for less-brittle tests.